mirage/unikernel.ml

```diff
@@ -348,6 +349,23 @@ module Main (N : Mirage_net.S) (ASSETS : Mirage_kv.RO) = struct
           Logs.err (fun m ->
               m "Got an error from %a:%d: %a" Ipaddr.pp ipaddr port pp_error err)
 
+    let to_map ~assoc m =
+      let open Multipart_form in
+      let rec go (map, rest) = function
+        | Leaf { header; body } -> (
+            match
+              Option.bind
+                (Header.content_disposition header)
+                Content_disposition.name
+            with
+            | Some name -> (Map.add name (None, List.assoc body assoc) map, rest)
+            | None -> (map, (body, (None, List.assoc body assoc)) :: rest))
+        | Multipart { body; _ } ->
+            let fold acc = function Some elt -> go acc elt | None -> acc in
+            List.fold_left fold (map, rest) body
```

Comment:
Why is there the local binding `let fold`, I'm used to have this directly as an argument to `List.fold_left`. Any particular reason?